src/front/glsl/functions.rs

```diff
@@ -281,4 +281,28 @@ impl Program {
         f.fill_global_use(self.module.global_variables.len(), &self.module.functions);
         f
     }
+
+    pub fn declare_function(&mut self, f: Function) {
+        let stage = if let Some(ref name) = f.name {
+            self.entry_points
```

Comment:
Can `entry_points` have the same name for multiple stages? I think IR would allow that, but GLSL is totally free to forbid this. If we don't allow it in GLSL, let's make `entry_points` a `FastHashMap<Name, ShaderStage>`?

Reply:
I think that would be weird in glsl, as in source that would be multiple functions with same name. Will change to `FastHashMap`